test/unit/interface_test.go

```diff
@@ -82,26 +157,33 @@ func TestInterface_Create(t *testing.T) {
 	base.MockPost("linode/instances/123/interfaces", fixtureData)
 
 	opts := linodego.LinodeInterfaceCreateOptions{
+		Public: &linodego.PublicInterfaceCreateOptions{
+			IPv4: &linodego.PublicInterfaceIPv4CreateOptions{
+				Addresses: []linodego.PublicInterfaceIPv4AddressCreateOptions{
+					{
+						Address: linodego.Pointer("auto"),
+						Primary: linodego.Pointer(true),
+					},
+				},
+			},
+		},
 		FirewallID: linodego.DoublePointer(123),
-		Public:     nil,
 	}
 
 	iface, err := base.Client.CreateInterface(context.Background(), 123, opts)
 	if err != nil {
 		t.Fatalf("Error fetching interfaces: %v", err)
 	}
 
-	assert.Equal(t, 123, iface.ID)
+	assert.Equal(t, 789, iface.ID)
 	assert.Equal(t, "auto", iface.Public.IPv4.Addresses[0].Address)
 }
 
-func TestInterface_Update(t *testing.T) {
```

Comment:
`TestInterface_Update` is redundant with the interface update tests introduced in this PR, so I removed it for now 🙂